src/services/date_utils.js

```diff
@@ -68,6 +68,43 @@ function getDateTimeForFile() {
     return new Date().toISOString().substr(0, 19).replace(/:/g, '');
 }
 
+/**
+ * Modify from https://stackoverflow.com/a/9047794
+ * Returns the week number for this date
+ */
+function getWeek(date, startOfTheWeek) {
```

Comment:
This returns ISO week, but I wonder if this is the best choice for "human consumption". For me, it's kinda surprising to see 1. 1. marked as "week 53 of the previous year".

Reply:
Yes, it's really odd but have to...

Reply:
But we don't have to use ISO weeks. There are other week numbering systems.

For me, it makes sense to say that January 1st is always the first week, no matter which day of the week it is. But I'm open to discussion on this point.

Reply:
Then you will feel strange that the Monday of 2023-W1 is 2022-12-26 >_<

Reply:
Not necessarily:

* January 1st is the 1. week (no matter which weekday it falls on)
* December 31st is always 53rd week

It has the disadvantage that first and last week in the year may not have 7 days (may be as short as 1 day), but that's IMHO acceptable.

Reply:
IMHO, I don't think a week which has less than 7 days is a good idea. This maybe need a discussion with more people?

Reply:
> IMHO, I don't think a week which has less than 7 days is a good idea. This maybe need a discussion with more people?

It's certainly not great, but personally, I don't think it's that bad if it affects only the days around the new year.

I agree that input from other people would be good.

Reply:
Personally speaking, I don't care about the week numbering(or week note id), because I can always change the title of the week notes. But I think one week should have 7 days.

In our daily life, we most likely use this week/next week/..., and don't combine "week" with "year". Therefore , I don't think use ISO week is a problem.

But I still feel that "year/week/day" is not a good structure for weekly.

Reply:
How about this idea? 

When the first day of year is not the start of week, for example 2023, we create a week note titled "W53 | W1" which will appear under both "2022" and "2023", and the "2023"'s note is a cloned note.

Reply:
@AllanZyne leveraging cloning actually sounds like a good idea. Reflecting on the previous discussion, I think the biggest problem with the ISO numbering was that the first days of the year would hierarchically appear under the preceding year, but this solution would fix this. So I would say, yes, let's do it like this.

Reply:
@zadam Great! I'll implement it ASAP.